### ISSUE TITLE
The performace is slow.

### ISSUE THREAD

**Issue by Oceania2018** [irmen/pickle#13]
W're trying to load a imdb dataset which is packed in pickle format.
The loading sample is in the repo of [TensorFlow.Keras](https://github.com/SciSharp/TensorFlow.NET/blob/c814fe121a77c90ca4d64f96faca6a98f99a93db/src/TensorFlowNET.Keras/Datasets/Imdb.cs#L78).

The specific slowness code is:
```csharp
(NDArray, NDArray) LoadX(byte[] bytes)
{
    var x = np.Load_Npz<int[,]>(bytes);
    return (x["x_train.npy"], x["x_test.npy"]);
}
```

You can use thie code snippet to reproduce this performance issue:
```csharp
var cfg = new TransformerClassificationConfig();
var dataloader = new IMDbDataset(cfg); //the dataset is initially downloaded at TEMP dir, e.g., C:\Users\{user name}\AppData\Local\Temp\imdb\imdb.npz
var dataset = dataloader.GetData();
```

**Comment by irmen** [comment]
Questions:

- I see no unpickling being called in the code?
- how large is the dataset?
- define "slow" more precisely. For example, how long does it take python itself to unpickle the data and how long does your code take.

**Comment by Oceania2018** [comment]
When we use the reproduce code sample above, will step into this [function](ReadObjectMatrix) (click to jump to the source code):
```csharp
Array ReadObjectMatrix(BinaryReader reader, Array matrix, int[] shape)
{
    Stream stream = reader.BaseStream;
    var unpickler = new Unpickler();
    var unpickledData = unpickler.load(stream);
    return (MultiArrayPickleWarpper)unpickledData;
}
```

The dataset can be found at https://ai.stanford.edu/~amaas/data/sentiment/.
It contains 25K sentences, not large, but to load this dataset, it took about 2 minutes.
![image](https://github.com/irmen/pickle/assets/1705364/25bf0932-5e3a-4a42-9da4-4c099dd18211)

I'll include the Razorvine.Pickle source code, to identify which part is the root cause.

Updated:
Most consuming time is here:
![image](https://github.com/irmen/pickle/assets/1705364/fb1fba16-00e3-46bc-b56d-8c5b810d51a3)

I found it for loop every 2 bytes (for int2), the dataset is a list of string.
Is there any way to optimize this for list of string without for loop the every bytes?

I saw there is `load_binstring` method, but not sure how to use it.
